src/otx/algo/segmentation/model/utils/utils.py

```diff
@@ -0,0 +1,449 @@
+"""Utils used for LiteHRNet model."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+from __future__ import annotations
+
+from typing import Callable, ClassVar
+
+import torch
+import torch.nn.functional as f
+from mmcv.cnn import ConvModule, DepthwiseSeparableConvModule
+from torch import nn
+from torch.nn import AdaptiveAvgPool2d, AdaptiveMaxPool2d
+
+
+def channel_shuffle(x: torch.Tensor, groups: int) -> torch.Tensor:
+    """Channel Shuffle operation.
+
+    This function enables cross-group information flow for multiple groups
+    convolution layers.
+
+    Args:
+        x (Tensor): The input tensor.
+        groups (int): The number of groups to divide the input tensor in the channel dimension.
+
+    Returns:
+        Tensor: The output tensor after channel shuffle operation.
+    """
+    batch_size, num_channels, height, width = x.size()
+    channels_per_group = num_channels // groups
+
+    x = x.view(batch_size, groups, channels_per_group, height, width)
+    x = torch.transpose(x, 1, 2).contiguous()
+
+    return x.view(batch_size, -1, height, width)
+
+class PSPModule(nn.Module):
+    """PSP module.
+
+    Reference: https://github.com/MendelXu/ANN.
+    """
+    methods: ClassVar[dict[str, AdaptiveMaxPool2d | AdaptiveAvgPool2d]] = {"max": AdaptiveMaxPool2d,
+                                                                           "avg": AdaptiveAvgPool2d}
+
+    def __init__(self, sizes: tuple = (1, 3, 6, 8), method: str = "max"):
+        super().__init__()
+
+        pool_block = self.methods[method]
+
+        self.stages = nn.ModuleList([pool_block(output_size=(size, size)) for size in sizes])
+
+    def forward(self, feats: torch.Tensor) -> torch.Tensor:
+        """Forward."""
+        batch_size, c, _, _ = feats.size()
+
+        priors = [stage(feats).view(batch_size, c, -1) for stage in self.stages]
+
+        return torch.cat(priors, -1)
+
+# pylint: disable=too-many-instance-attributes
```

Comment:
Need to remove

Reply:
Same here: https://github.com/openvinotoolkit/training_extensions/pull/2693/files#diff-fff119407ff45ba5eca03ce184fec51f68e89857803b4b7c46473f94e7fee345R178

Reply:
removed [29dd2eb](https://github.com/openvinotoolkit/training_extensions/pull/2693/commits/29dd2eb8810801d217caef839453df331d47e086)